merkletree/proof.go

```diff
@@ -16,6 +23,75 @@ type ProofNode interface {
 	Commitment() []byte
 }
 
+func computeLeafHash(ap *AuthenticationPath,
+	leafIndex, leafCommitment []byte, leafLevel int, isLeafEmpty bool) (leafHash []byte) {
+	if isLeafEmpty {
```

Comment:
> If we're passing in ap to computeLeafHash() couldn't we use the index, commitment, level etc from the ap.Leaf directly inside the function?

I think we cannot use it since we have to instantiate new `AuthenticationPath` from input arrays (https://github.com/coniks-sys/coniks-go/blob/7801e29c2bbee6962ce9f2fb6b9fac583bc66633/client/cgo/coniks.go#L106-L109), and `ap.Leaf` is just an interface instance. 

> On the other hand, I think this would be a similar issue to what we discussed in d46c806#commitcomment-18437585, so I don't know if we should be passing in any auth path pointer at all.

We didn't do any serialization so I think it's okay to pass an auth path pointer, instead passing a lot of arrays.